cmd/vcert/result_writer.go

```diff
@@ -117,6 +122,106 @@ func (o *Output) AsPKCS12(c *Config) ([]byte, error) {
 	return bytes, nil
 }
 
+func (o *Output) AsJKS(c *Config) ([]byte, error) {
+
+	var err interface{}
+
+	if len(o.Certificate) == 0 || len(o.PrivateKey) == 0 {
+		return nil, fmt.Errorf("at least certificate and private key are required")
+	}
+
+	var certificateChain []keystore.Certificate
+
+	//getting the certificate in bytes
+	p, _ := pem.Decode([]byte(o.Certificate))
+	if p == nil || p.Type != "CERTIFICATE" {
+		return nil, fmt.Errorf("certificate parse error(1)")
+	}
+	var certInBytes = p.Bytes
+
+	//adding the certificates to the slice of Certificates
+	certificateChain = append(certificateChain, keystore.Certificate{
+		Type:    "X509",
+		Content: certInBytes,
+	})
+
+	// getting each one of the certificates in the chain of certificates and adding their bytes to the chain of certificates
+	for _, chainCert := range o.Chain {
+		crt, _ := pem.Decode([]byte(chainCert))
+		certificateChain = append(certificateChain, keystore.Certificate{
+			Type:    "X509",
+			Content: crt.Bytes,
+		})
+	}
+
+	// getting the bytes of the PK
+	p, _ = pem.Decode([]byte(o.PrivateKey))
+	if p == nil {
+		return nil, fmt.Errorf("missing private key PEM")
+	}
+	var privDER []byte
+
+	//decrypting the PK because due the restriction that always will be requested the key password
+	//to the user(--key-password or pass phrase value from prompt) for jks format then the PK always
+	//will be encrypted with the key password provided
+	privDER, err = x509.DecryptPEMBlock(p, []byte(c.KeyPassword))
+	if err != nil {
+		return nil, fmt.Errorf("private key PEM decryption error: %s", err)
+	}
+
+	//Unmarshalling the PK
+	var privKey interface{}
```

Comment:
The same with privKey, declaration not needed

Reply:
the same answer as the previous one :-P